src/main/java/org/openrewrite/java/spring/http/SimplifyMediaTypeParseCalls.java

```diff
@@ -40,22 +40,21 @@ public TreeVisitor<?, ExecutionContext> getVisitor() {
     }
 
     private static final class SimplifyParseCallsVisitor extends JavaVisitor<ExecutionContext> {
-        private final MethodMatcher MEDIATYPE_PARSE_MATCHER = new MethodMatcher("org.springframework.http.MediaType parse(..)");
+        private final MethodMatcher MEDIATYPE_PARSE_MEDIA_TYPE_MATCHER = new MethodMatcher("org.springframework.http.MediaType parseMediaType(String)");
+        private final MethodMatcher MEDIATYPE_VALUE_OF_MATCHER = new MethodMatcher("org.springframework.http.MediaType valueOf(String)");
 
         @Override
         public J visitMethodInvocation(J.MethodInvocation mi, ExecutionContext ctx) {
-            if (!MEDIATYPE_PARSE_MATCHER.matches(mi) && !mi.getArguments().isEmpty()) {
-                return mi;
+            if (MEDIATYPE_PARSE_MEDIA_TYPE_MATCHER.matches(mi) || MEDIATYPE_VALUE_OF_MATCHER.matches(mi)) {
+                J.Literal test = (J.Literal) mi.getArguments().get(0);
+                if ("application/json".equals(test.getValue())) {
+                    return JavaTemplate.builder("MediaType.APPLICATION_JSON")
```

Comment:
This is perhaps too limiting; we already have a recipe to replace literals with constants:
- https://github.com/openrewrite/rewrite-spring/issues/325

A comment there was logged as an improvement to pick up cases where we (after the above recipe) pass a constant into a parse method:
- https://github.com/openrewrite/rewrite-spring/issues/337

If you ask me this implementation should detect String constants passed into parse, and replace those with the pre existing MediaType constants. Open to discuss.

Reply:
Ok so just to make sure I understand this correctly instead of detecting something like `"application/json"` like we are doing now; We should instead detect constants and make the replacement then?

Are there are other `MediaType` constants this recipe should try covering also? Instead of just looking for `APPLICATION_JSON`?

Reply:
yes when the value feeding into the parse call is a constant defined on `MediaType` then you want to replace the whole method invocation with the sister constant.

Reply:
so
```diff
-MediaType.parseMediaType(MediaType.APPLICATION_JSON_VALUE)
+MediaType.APPLICATION_JSON
```
notice how we drop the method call, and replace it with a "sister constant" that lacks the `_VALUE` suffix.

Reply:
the above is from memory, so be sure to check what's available on the MediaType class in your IDE, but that's the rough outline of what I would expect. And we should not limit ourselves to `APPLICATION_JSON`, instead just any constants that follow that pattern (without naming each explicitly).